src/main/java/org/opensearch/performanceanalyzer/writer/EventLogQueueProcessor.java

```diff
@@ -97,7 +114,19 @@ public void scheduleExecutor() {
 
     // This executes every purgePeriodicityMillis interval.
     public void purgeQueueAndPersist() {
-        // Return if the writer is not enabled.
+        counter += 1;
+        long currentTimeMillis = System.currentTimeMillis();
+
+        // Delete the older event log files every filesCleanupPeriod (defaults to 60)
+        // In case files deletion takes longer/fails, we are okay with eventQueue reaching
+        // its max size (100000), post that {@link PerformanceAnalyzerMetrics#emitMetric()}
+        // will emit metric {@link WriterMetrics#METRICS_WRITE_ERROR} and return.
+        if (counter >= filesCleanupPeriod) {
+            eventLogFileHandler.deleteFiles(currentTimeMillis, filesCleanupPeriodicityMillis);
+            counter = 0;
+        }
+
+        // Drain the Queue, and if writer is enabled then persist to event log file.
```

Comment:
I would also do emission of metrics and then do the cleanup, so that the time budget for cleanup comes from the time the thread would have slept anyways.

Reply:
Good point . I have extracted this logic out to a separate function `cleanup()` and invoking it post the write to disk complete.